hdfs/src/main/scala/akka/stream/alpakka/hdfs/model.scala

```diff
@@ -0,0 +1,166 @@
+final case class IncomingMessage[T, C](source: T, passThrough: C)
+
+object IncomingMessage {
+  // Apply method to use when not using passThrough
+  def apply[T](source: T): IncomingMessage[T, NotUsed] =
+    IncomingMessage(source, NotUsed)
+
+  // Java-api - without passThrough
+  def create[T](source: T): IncomingMessage[T, NotUsed] =
+    IncomingMessage(source)
+
+  // Java-api - with passThrough
+  def create[T, C](source: T, passThrough: C): IncomingMessage[T, C] =
+    IncomingMessage(source, passThrough)
+
+}
+
+sealed abstract class OutgoingMessage[+T]
+final case class RotationMessage(path: String, rotation: Int) extends OutgoingMessage[Nothing]
+final case class WrittenMessage[T](passThrough: T, inRotation: Int) extends OutgoingMessage[T]
+
+sealed case class FileUnit(byteCount: Long)
+
+object FileUnit {
+  val KB = FileUnit(Math.pow(2, 10).toLong)
+  val MB = FileUnit(Math.pow(2, 20).toLong)
+  val GB = FileUnit(Math.pow(2, 30).toLong)
+  val TB = FileUnit(Math.pow(2, 40).toLong)
+}
+
+sealed abstract class FilePathGenerator extends ((Long, Long) => Path) {
+  def tempDirectory: String
+}
+
+object FilePathGenerator {
+  private val DefaultTempDirectory = "/tmp/alpakka-hdfs"
+
+  /**
+   * Scala API: creates [[FilePathGenerator]] to rotate output
+   *
+   * @param f    a function that takes rotation count and timestamp to return path of output
+   * @param temp the temporary directory that [[akka.stream.alpakka.hdfs.impl.HdfsFlowStage]] use
+   */
+  def apply(f: (Long, Long) => String, temp: String = DefaultTempDirectory): FilePathGenerator =
+    new FilePathGenerator {
+      val tempDirectory: String = temp
+      def apply(rotationCount: Long, timestamp: Long): Path = new Path(f(rotationCount, timestamp))
+    }
+
+  /**
+   * Java API: creates [[FilePathGenerator]] to rotate output
+   *
+   * @param f a function that takes rotation count and timestamp to return path of output
+   */
+  def create(f: BiFunction[java.lang.Long, java.lang.Long, String]): FilePathGenerator =
+    FilePathGenerator(javaFuncToScalaFunc(f), DefaultTempDirectory)
+
+  /**
+   * Java API: creates [[FilePathGenerator]] to rotate output
+   *
+   * @param f a function that takes rotation count and timestamp to return path of output
+   */
+  def create(f: BiFunction[java.lang.Long, java.lang.Long, String], temp: String): FilePathGenerator =
+    FilePathGenerator(javaFuncToScalaFunc(f), temp)
+
+  private def javaFuncToScalaFunc(f: BiFunction[java.lang.Long, java.lang.Long, String]): (Long, Long) => String =
+    (rc, t) => f.apply(rc, t)
+
+}
+
+abstract class RotationStrategy extends Strategy {
+  type S = RotationStrategy
+}
+
+object RotationStrategy {
+
+  /**
+   * Creates [[SizeRotationStrategy]]
+   *
+   * @param count a count of [[FileUnit]]
+   * @param unit a file unit
+   */
+  def size(count: Double, unit: FileUnit): RotationStrategy =
+    SizeRotationStrategy(0, count * unit.byteCount)
+
+  /**
+   * Creates [[CountRotationStrategy]]
```

Comment:
The docs would try to link to private classes, which are not included in the scaladocs.
Describe the strategy instead.